containers/unit_tests/TestDynRankViewTypedefs.cpp

```diff
@@ -180,93 +181,74 @@ constexpr bool test_view_typedefs_impl() {
 template<class T, class ... ViewArgs>
 struct ViewParams {};
 
-template<class L, class S, class M, class HostMirrorSpace, class ValueType, class ReferenceType, class T, class ... ViewArgs>
+template<class L, class S, class M, class HostMirrorDevice, class ValueType, class ReferenceType, class T, class ... ViewArgs>
 constexpr bool test_view_typedefs(ViewParams<T, ViewArgs...>) {
   return test_view_typedefs_impl<Kokkos::DynRankView<T, ViewArgs...>, Kokkos::ViewTraits<T*******, ViewArgs...>,
-                                 T, L, S, M, HostMirrorSpace, ValueType, ReferenceType>();
+                                 T, L, S, M, HostMirrorDevice, ValueType, ReferenceType>();
 }
 
-
-constexpr bool is_host_exec = std::is_same_v<Kokkos::DefaultExecutionSpace, Kokkos::DefaultHostExecutionSpace>;
-
-#if defined(KOKKOS_ENABLE_CUDA_UVM) || defined(KOKKOS_ENABLE_IMPL_CUDA_UNIFIED_MEMORY) || defined(KOKKOS_IMPL_HIP_UNIFIED_MEMORY)
-constexpr bool has_unified_mem_space = true;
-#else
-constexpr bool has_unified_mem_space = false;
-#endif
-
 // The test take explicit template arguments for: LayoutType, Space, MemoryTraits, HostMirrorSpace, ValueType, ReferenceType
 // The ViewParams is just a type pack for the View template arguments
 
 // Kokkos::View<int>
 namespace TestInt {
-  using layout_type = Kokkos::DefaultExecutionSpace::array_layout;
-  using space = Kokkos::DefaultExecutionSpace;
+  using layout_type   = Kokkos::DefaultExecutionSpace::array_layout;
   using memory_traits = Kokkos::MemoryTraits<0>;
-  // HostMirrorSpace is a mess so: if the default exec is a host exec, that is it
-  using host_mirror_space = std::conditional_t<is_host_exec, Kokkos::DefaultExecutionSpace,
-  // otherwise if unified memory is not on its HostSpace
-                               std::conditional_t<!has_unified_mem_space, Kokkos::HostSpace,
-  // otherwise its the following Device type
-                               Kokkos::Device<Kokkos::DefaultHostExecutionSpace, typename Kokkos::DefaultExecutionSpace::memory_space>>>;
-  static_assert(test_view_typedefs<layout_type, space, memory_traits, host_mirror_space, int, int&>(
+  using space         = Kokkos::DefaultExecutionSpace;
+  using device        = Kokkos::Device<space, typename space::memory_space>;
+  using host_mirror_device = Kokkos::Impl::HostMirror<device>::Device;
```

Comment:
The test previously very explicitly did not use the internal pieces of Kokkos to figure out what the expected host type is. This was done in order to make sure our internals do what we expect. Right now a bug in Kokkos::Impl::HostMirror would make the test pass, since both the View and the test use the same implementation. So say `HostMirror::Device` as `Device<Cuda, CudaSpace>` would make the test still pass even though the implementation is faulty.

Reply:
Yep, restoring the original  test / code.